client/components/data/domain-management/index.jsx

```diff
@@ -1,101 +1,240 @@
+		}
+		if ( needsEmailForwarding ) {
+			stores.push( EmailForwardingStore );
+		}
+		if ( needsNameservers ) {
+			stores.push( NameserversStore );
+		}
+		if ( needsSiteRedirect ) {
+			stores.push( SiteRedirectStore );
+		}
+		if ( needsUsers ) {
+			stores.push( UsersStore );
+		}
+		if ( needsWhois ) {
+			stores.push( WhoisStore );
 		}
-	},
 
-	render: function() {
 		return (
 			<div>
 				<PageViewTracker path={ this.props.analyticsPath } title={ this.props.analyticsTitle } />
+				{ selectedSite && needsContactDetails && <QueryContactDetailsCache /> }
+				{ selectedSite && needsDomains && <QuerySiteDomains siteId={ selectedSite.ID } /> }
+				{ selectedSite && needsPlans && <QuerySitePlans siteId={ selectedSite.ID } /> }
+				{ needsProductsList && <QueryProductsList /> }
+
 				<StoreConnection
 					component={ this.props.component }
-					stores={ stores }
+					context={ this.props.context }
+					currentUser={ this.props.currentUser }
+					domains={ this.props.domains }
 					getStateFromStores={ getStateFromStores }
-					products={ this.props.productsList.get() }
+					googleAppsUsers={ this.props.googleAppsUsers }
+					googleAppsUsersLoaded={ this.props.googleAppsUsersLoaded }
+					isRequestingSiteDomains={ this.props.isRequestingSiteDomains }
+					products={ this.props.productsList }
 					selectedDomainName={ this.props.selectedDomainName }
-					selectedSite={ this.props.selectedSite }
+					selectedSite={ selectedSite }
 					sitePlans={ this.props.sitePlans }
-					context={ this.props.context }
+					stores={ stores }
 				/>
-				{ this.props.selectedSite && <QuerySitePlans siteId={ this.props.selectedSite.ID } /> && (
-						<QueryContactDetailsCache />
-					) }
 			</div>
 		);
-	},
-} );
+	}
+}
 
-const mapStateToProps = state => {
-	const selectedSite = getSelectedSite( state );
+export default connect(
+	( state, { selectedDomainName } ) => {
+		const selectedSite = getSelectedSite( state );
+		const siteId = get( selectedSite, 'ID', null );
+		const googleAppsUsers = selectedDomainName
+			? getByDomain( state, selectedDomainName )
+			: getBySite( state, selectedSite.ID );
 
-	return {
-		sitePlans: getPlansBySite( state, selectedSite ),
-		selectedSite,
-	};
-};
+		return {
+			currentUser: getCurrentUser( state ),
+			domains: getDecoratedSiteDomains( state, siteId ),
+			googleAppsUsers,
+			googleAppsUsersLoaded: isLoaded( state ),
+			isRequestingSiteDomains: isRequestingSiteDomains( state, siteId ),
+			productsList: getProductsList( state ),
+			sitePlans: getPlansBySite( state, selectedSite ),
+			selectedSite,
+		};
+	},
+	( dispatch, { selectedDomainName } ) => {
+		const googleAppsUsersFetcher = selectedDomainName
+			? () => fetchByDomain( selectedDomainName )
+			: siteId => fetchBySiteId( siteId );
 
-export default connect( mapStateToProps )( DomainManagementData );
+		return {
+			fetchGoogleAppsUsers: siteId => dispatch( googleAppsUsersFetcher( siteId ) ),
```

Comment:
i think this is going to be a new function on every run of `connect`, which means the component will rerender...

@jsnajdr do I have that right?